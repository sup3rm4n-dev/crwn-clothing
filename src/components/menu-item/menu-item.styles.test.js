import React from 'react';
import renderer from 'react-test-renderer';
import {
  MenuItemContainer,
  BackgroundImageContainer,
} from './menu-item.styles';

describe('should render menu-item styled components', () => {
  describe('MenuItemContainer styled-component', () => {
    const tree = renderer.create(<MenuItemContainer />).toJSON();

    it('should render MenuItemContainer without props', () => {
      expect(tree).toMatchSnapshot();
    });

    it('should have a height of 24rem', () => {
      expect(tree).toHaveStyleRule('height', '24rem');
    });
  });

  describe('MenuItemContainer styled-component with size prop', () => {
    const tree = renderer.create(<MenuItemContainer size />).toJSON();

    it('should render MenuItemContainer with size prop', () => {
      expect(tree).toMatchSnapshot();
    });

    it('should have a height of 38rem', () => {
      expect(tree).toHaveStyleRule('height', '38rem');
    });
  });

  describe('BackgroundImageContainer styled-component', () => {
    const tree = renderer.create(<BackgroundImageContainer />).toJSON();

    it('should render BackgroundImageContainer without props', () => {
      expect(tree).toMatchSnapshot();
    });

    it('should have a background-image url undefined', () => {
      expect(tree).toHaveStyleRule('background-image', 'url(undefined)');
    });
  });

  describe('BackgroundImageContainer styled-component with imageUrl prop', () => {
    const tree = renderer
      .create(
        <BackgroundImageContainer imageUrl='http://example.com/testimage.jpeg' />
      )
      .toJSON();

    it('should render BackgroundImage with imageUrl prop', () => {
      expect(tree).toMatchSnapshot();
    });

    it('should have a background-image url style rule equal to imageUrl prop', () => {
      expect(tree).toHaveStyleRule(
        'background-image',
        'url(http://example.com/testimage.jpeg)'
      );
    });
  });
});
