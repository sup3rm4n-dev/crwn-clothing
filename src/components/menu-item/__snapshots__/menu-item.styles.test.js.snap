// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render menu-item styled components BackgroundImageContainer styled-component should render BackgroundImageContainer without props 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  background-position: center;
  background-size: cover;
  background-image: url(undefined);
}

<div
  className="c0"
/>
`;

exports[`should render menu-item styled components BackgroundImageContainer styled-component with imageUrl prop should render BackgroundImage with imageUrl prop 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  background-position: center;
  background-size: cover;
  background-image: url(http://example.com/testimage.jpeg);
}

<div
  className="c0"
/>
`;

exports[`should render menu-item styled components MenuItemContainer styled-component should render MenuItemContainer without props 1`] = `
.c0 {
  height: 240px;
  min-width: 30%;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border: 1px solid black;
  margin: 0 7.5px 15px;
  overflow: hidden;
}

.c0:hover {
  cursor: pointer;
}

.c0:hover .background-image {
  -webkit-transform: scale(1.11);
  -ms-transform: scale(1.11);
  transform: scale(1.11);
  -webkit-transition: -webkit-transform 6.4s cubic-bezier(0.1,0.26,0.41,0.87) 38ms;
  -webkit-transition: transform 6.4s cubic-bezier(0.1,0.26,0.41,0.87) 38ms;
  transition: transform 6.4s cubic-bezier(0.1,0.26,0.41,0.87) 38ms;
}

.c0:hover .content {
  opacity: 0.7;
}

.c0:first-child {
  margin-right: 7.5px;
}

.c0:last-child {
  margin-left: 7.5px;
}

@media screen and (max-width:800px) {
  .c0 {
    height: 200px;
  }
}

<div
  className="c0"
/>
`;

exports[`should render menu-item styled components MenuItemContainer styled-component with size prop should render MenuItemContainer with size prop 1`] = `
.c0 {
  height: 380px;
  min-width: 30%;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border: 1px solid black;
  margin: 0 7.5px 15px;
  overflow: hidden;
}

.c0:hover {
  cursor: pointer;
}

.c0:hover .background-image {
  -webkit-transform: scale(1.11);
  -ms-transform: scale(1.11);
  transform: scale(1.11);
  -webkit-transition: -webkit-transform 6.4s cubic-bezier(0.1,0.26,0.41,0.87) 38ms;
  -webkit-transition: transform 6.4s cubic-bezier(0.1,0.26,0.41,0.87) 38ms;
  transition: transform 6.4s cubic-bezier(0.1,0.26,0.41,0.87) 38ms;
}

.c0:hover .content {
  opacity: 0.7;
}

.c0:first-child {
  margin-right: 7.5px;
}

.c0:last-child {
  margin-left: 7.5px;
}

@media screen and (max-width:800px) {
  .c0 {
    height: 200px;
  }
}

<div
  className="c0"
  size={true}
/>
`;
